src/main/java/org/magicdgs/readtools/tools/mappability/gem/GemMappabilityToBed.java

```diff
@@ -0,0 +1,129 @@
+ *     <li>
+ *         Sequences in FASTA-like format: sequence name preceded by ~ and a character per-base
+ *         encoded in the header. Each character encodes a different range of values,
+ *         which represent the number of mappings of the reads starting at that position.
+ *     </li>
+ * </ul>
+ *
+ * <p>This tool parses the GEM-mappability file and outputs a per-position BED-graph format. For
+ * each position, different scoring systems could be use (see arguments for more information).
+ * </p>
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+@DocumentedFeature
+@CommandLineProgramProperties(oneLineSummary = "Converts a GEM-mappability file into a BED-graph format",
+        summary = GemMappabilityToBed.SUMMARY,
+        programGroup = MappabiltityProgramGroup.class
+)
+@ExperimentalFeature
+public final class GemMappabilityToBed extends ReadToolsProgram {
+
+    protected static final String SUMMARY = "Converts a GEM-mappability file (Derrien et al. 2012) "
+            + "into a BED-graph format converting the number of mappings to different kind of scores.\n"
+            + "The GEM-mappability format is a FASTA-like file with:\n"
+            + "\t- Header including information for the algorithm applied and encoding.\n"
+            + "\t- Sequences in a FASTA-like format representing number of mappings\n"
+            + "Each character encodes a different range of values, which represent the number of "
+            + "mappings of each base."
+            + "\n\n"
+            + "Find more information about this tool in "
+            + RTHelpConstants.DOCUMENTATION_PAGE + "GemMappabilityToBed.html";
+
+    @Argument(fullName = RTStandardArguments.INPUT_LONG_NAME, shortName = RTStandardArguments.INPUT_SHORT_NAME, doc = "GEM-mappability file (FASTA-like) to parse")
+    public String input;
+
+    /**
+     * Bed-graph output with the number of mappings as score. If it contains a block-compressed
+     * extension (e.g., .bgz), it will be compressed.
+     */
+    @Argument(fullName = RTStandardArguments.OUTPUT_LONG_NAME, shortName = RTStandardArguments.OUTPUT_SHORT_NAME, doc = "BED-graph output with the number of mappings as score")
+    public String output;
+
+    @Argument(fullName = "score-method", doc = "Method to convert number of mappings within a range into the reported score")
+    public GemScoreMethod method = GemScoreMethod.MID;
+
+    private GemMappabilityReader reader;
+    private PrintStream writer;
+
+    @Override
+    protected void onStartup() {
+        final Path inputPath = IOUtils.getPath(input);
+        try {
+            reader = new GemMappabilityReader(inputPath);
+        } catch (final IOException e) {
+            throw new UserException.CouldNotReadInputFile(inputPath, e);
+        }
+        final Path outputPath = IOUtils.getPath(output);
+        try {
+            // TODO: should support arbitrary compressed output (as other tools)
```

Comment:
Is there an issue for this already?

Reply:
Created https://github.com/magicDGS/ReadTools/issues/493 for all the work that has to be done in this regard